cmd/collectors/zapi/plugins/volume/volume.go

```diff
@@ -90,13 +97,26 @@ func (my *Volume) Run(data *matrix.Matrix) ([]*matrix.Matrix, error) {
 		my.currentVal = 0
 
 		// invoke snapmirror zapi and populate info in source and destination snapmirror maps
-		smSourceMap, smDestinationMap, err := my.GetSnapMirrors()
-		if err != nil {
-			my.Logger.Error().Stack().Err(err).Msg("Failed to collect snapmirror data")
+		if smSourceMap, smDestinationMap, err := my.GetSnapMirrors(); err != nil {
+			my.Logger.Warn().Stack().Err(err).Msg("Failed to collect snapmirror data")
+		} else {
+			// update internal cache based on volume and SM maps
+			my.updateMaps(data, smSourceMap, smDestinationMap)
 		}
 
-		// update internal cache based on volume and SM maps
-		my.updateMaps(data, smSourceMap, smDestinationMap)
+		// invoke disk-encrypt-get-iter zapi and populate disk info
+		disks, err1 := my.getDiskData()
+		// invoke aggr-status-get-iter zapi and populate aggr disk mapping info
+		aggrDiskMap, err2 := my.getAggrDiskMapping()
+
+		if err1 != nil {
+			my.Logger.Warn().Stack().Err(err1).Msg("Failed to collect disk data")
+		}
+		if err2 != nil {
+			my.Logger.Warn().Stack().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
```

Comment:
does it make sense to continue to line 119 if there is an err?

Reply:
yes, function updateAggrMap at line 119 can handle this.